### ISSUE TITLE
syntax warnings on .rst files while 'make html'

### ISSUE THREAD

**Issue by 9bow** [pytorch/tutorials#393]
Some reST syntax warnings occur while building HTML.

Warnings that occur are as below:
```
...
reading sources... [100%] intermediate/spatial_transformer_tutorial
/Users/reserve/Workspace/tutorials/advanced/cpp_extension.rst:2: WARNING: Duplicate explicit target name: "here".
/Users/reserve/Workspace/tutorials/advanced/cpp_extension.rst:2: WARNING: Duplicate explicit target name: "here".
/Users/reserve/Workspace/tutorials/advanced/torch_script_custom_ops.rst:23: WARNING: Inline literal start-string without end-string.
/Users/reserve/Workspace/tutorials/advanced/torch_script_custom_ops.rst:1010: WARNING: Error in "code-block" directive:
1 argument(s) required, 0 supplied.

.. code-block::

  from setuptools import setup
  from torch.utils.cpp_extension import BuildExtension, CppExtension

  setup(
      name="warp_perspective",
      ext_modules=[
          CppExtension(
              "warp_perspective",
              ["example_app/warp_perspective/op.cpp"],
              libraries=["opencv_core", "opencv_imgproc"],
          )
      ],
      cmdclass={"build_ext": BuildExtension.with_options(no_python_abi_suffix=True)},
  )
/Users/reserve/Workspace/tutorials/advanced/torch_script_custom_ops.rst:1073: WARNING: Error in "code-block" directive:
1 argument(s) required, 0 supplied.

.. code-block::

  >>> import torch
  >>> torch.ops.load_library("warp_perspective.so")
  >>> print(torch.ops.custom.warp_perspective)
  <built-in method custom::warp_perspective of PyCapsule object at 0x7ff51c5b7bd0>
/Users/reserve/Workspace/tutorials/advanced/torch_script_custom_ops.rst:1088: WARNING: Unknown target name: "here<https://github.com/pytorch/pytorch/blob/master/aten/src/aten/core/jit_type.h>".
/Users/reserve/Workspace/tutorials/beginner/audio_classifier_tutorial.rst:11: WARNING: Duplicate explicit target name: "here".
/Users/reserve/Workspace/tutorials/beginner/blitz/cifar10_tutorial.rst:132: WARNING: Title underline too short.

2. Define a Convolutional Neural Network
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/reserve/Workspace/tutorials/beginner/blitz/cifar10_tutorial.rst:132: WARNING: Title underline too short.

2. Define a Convolutional Neural Network
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/reserve/Workspace/tutorials/beginner/nn_basics.rst:415: WARNING: Unexpected indentation.
/Users/reserve/Workspace/tutorials/beginner/nn_basics.rst:427: WARNING: Unexpected indentation.
/Users/reserve/Workspace/tutorials/beginner/nn_basics.rst:526: WARNING: Unexpected indentation.
/Users/reserve/Workspace/tutorials/beginner/nn_basics.rst:532: WARNING: Unexpected indentation.
/Users/reserve/Workspace/tutorials/beginner/nn_basics.rst:615: WARNING: Unexpected indentation.
/Users/reserve/Workspace/tutorials/beginner/nn_basics.rst:621: WARNING: Unexpected indentation.
/Users/reserve/Workspace/tutorials/beginner/nn_basics.rst:665: WARNING: Unexpected indentation.
/Users/reserve/Workspace/tutorials/beginner/nn_basics.rst:671: WARNING: Unexpected indentation.
looking for now-outdated files... none found
pickling environment... done
checking consistency... /Users/reserve/Workspace/tutorials/beginner/audio_classifier_tutorial.rst: WARNING: document isn't included in any toctree
/Users/reserve/Workspace/tutorials/beginner/former_torchies_tutorial.rst: WARNING: document isn't included in any toctree
/Users/reserve/Workspace/tutorials/beginner/hybrid_frontend_tutorial.rst: WARNING: document isn't included in any toctree
/Users/reserve/Workspace/tutorials/beginner/nn_basics.rst: WARNING: document isn't included in any toctree
/Users/reserve/Workspace/tutorials/beginner/ptcheat.rst: WARNING: document isn't included in any toctree
done
...
```

**Comment by jlin27** [comment]
@9bow  - Are you still getting these warnings?

**Comment by 9bow** [comment]
@jlin27 Some have been fixed, but new things happen.

Below lines are what I get while building tutorials via `make html-noplot` command with a fresh copy of this repo.

```
/path/to/pytorch/tutorials/advanced/cpp_extension.rst:2: WARNING: Duplicate explicit target name: "here".
/path/to/pytorch/tutorials/advanced/cpp_extension.rst:2: WARNING: Duplicate explicit target name: "here".
/path/to/pytorch/tutorials/advanced/cpp_frontend.rst:2: WARNING: Duplicate explicit target name: "here".
/path/to/pytorch/tutorials/advanced/cpp_frontend.rst:2: WARNING: Duplicate explicit target name: "here".
/path/to/pytorch/tutorials/advanced/cpp_frontend.rst:2: WARNING: Duplicate explicit target name: "docs".
/path/to/pytorch/tutorials/advanced/super_resolution_with_onnxruntime.rst:11: WARNING: Title underline too short.

4. (optional) Exporting a Model from PyTorch to ONNX and Running it using ONNX Runtime
========================================================================
/path/to/pytorch/tutorials/advanced/torch_script_custom_ops.rst:23: WARNING: Inline literal start-string without end-string.
/path/to/pytorch/tutorials/beginner/Intro_to_TorchScript_tutorial.rst:11: WARNING: Title underline too short.

2. Introduction to TorchScript
===========================
/path/to/pytorch/tutorials/beginner/blitz/cifar10_tutorial.rst:76: WARNING: Unexpected indentation.
/path/to/pytorch/tutorials/beginner/blitz/cifar10_tutorial.rst:132: WARNING: Title underline too short.

2. Define a Convolutional Neural Network
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/path/to/pytorch/tutorials/beginner/blitz/cifar10_tutorial.rst:132: WARNING: Title underline too short.

2. Define a Convolutional Neural Network
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/path/to/pytorch/tutorials/beginner/nn_tutorial.rst:561: WARNING: Unexpected indentation.
/path/to/pytorch/tutorials/beginner/nn_tutorial.rst:573: WARNING: Unexpected indentation.
/path/to/pytorch/tutorials/beginner/nn_tutorial.rst:713: WARNING: Unexpected indentation.
/path/to/pytorch/tutorials/beginner/nn_tutorial.rst:719: WARNING: Unexpected indentation.
/path/to/pytorch/tutorials/beginner/nn_tutorial.rst:825: WARNING: Unexpected indentation.
/path/to/pytorch/tutorials/beginner/nn_tutorial.rst:831: WARNING: Unexpected indentation.
/path/to/pytorch/tutorials/beginner/nn_tutorial.rst:890: WARNING: Unexpected indentation.
/path/to/pytorch/tutorials/beginner/nn_tutorial.rst:896: WARNING: Unexpected indentation.
/path/to/pytorch/tutorials/beginner/nn_tutorial.rst:1267: WARNING: Definition list ends without a blank line; unexpected unindent.
/path/to/pytorch/tutorials/beginner/ptcheat.rst:33: WARNING: Title underline too short.

Torchscript and JIT
---------------
/path/to/pytorch/tutorials/beginner/ptcheat.rst:33: WARNING: Title underline too short.

Torchscript and JIT
---------------
/path/to/pytorch/tutorials/beginner/transformer_tutorial.rst:442: WARNING: Title underline too short.

Evaluate the model with the test dataset
-------------------------------------
/path/to/pytorch/tutorials/beginner/transformer_tutorial.rst:442: WARNING: Title underline too short.

Evaluate the model with the test dataset
-------------------------------------
/path/to/pytorch/tutorials/index.rst:262: WARNING: Title underline too short.

PyTorch in Other Languages
-------------
/path/to/pytorch/tutorials/index.rst:262: WARNING: Title underline too short.

PyTorch in Other Languages
-------------
/path/to/pytorch/tutorials/intermediate/flask_rest_api_tutorial.rst:11: WARNING: Duplicate explicit target name: "here".
/path/to/pytorch/tutorials/intermediate/flask_rest_api_tutorial.rst:11: WARNING: Duplicate explicit target name: "this tutorial".
/path/to/pytorch/tutorials/intermediate/model_parallel_tutorial.rst:91: WARNING: Title underline too short.

Apply Model Parallel to Existing Modules
=======================
/path/to/pytorch/tutorials/intermediate/model_parallel_tutorial.rst:91: WARNING: Title underline too short.

Apply Model Parallel to Existing Modules
=======================
/path/to/pytorch/tutorials/intermediate/model_parallel_tutorial.rst:261: WARNING: Title underline too short.

Speed Up by Pipelining Inputs
=======================
/path/to/pytorch/tutorials/intermediate/model_parallel_tutorial.rst:261: WARNING: Title underline too short.

Speed Up by Pipelining Inputs
=======================
/path/to/pytorch/tutorials/intermediate/tensorboard_tutorial.rst:2: WARNING: Title underline too short.

Visualizing Models, Data, and Training with TensorBoard
====================================================
```

If you don't mind, I'll add commits to fix this on PR #394

**Comment by 9bow** [comment]
There're still many warning/errors on building whole tutorials, but the related PR is closed and tutorials also have many changes.
